pkg/virt-controller/services/template.go

```diff
@@ -798,12 +798,22 @@ func getResourceNameForNetwork(network *networkv1.NetworkAttachmentDefinition) s
 	return "" // meaning the network is not served by resources
 }
 
+func getNamespaceAndNetworkName(vmi *v1.VirtualMachineInstance, fullNetworkName string) (namespace string, networkName string) {
```

Comment:
This is still not enough. The full format of the network request is `namespace/network@ifname`. https://github.com/intel/multus-cni/blob/master/k8sclient/k8sclient.go#L151

Could you also handle the last ifname part? We don't need to use it as part of this PR, just not to assume network@ifname is network name.

Reply:
Actually, I'm not sure whether it could ever happen in VMI API. Perhaps we should block networkName that contains request for specific interface name.

Reply:
Picking a name of an interface in VMI scope seems like a weird request since the netlink implementation is internal to launcher container. While I agree it may make sense to forbid the format completely on api level it seems to me this is not in scope of the PR.